### PR TITLE
fix: Add clarification to Quarkus chapter

### PR DESCRIPTION
**Pull request by mshabarov**
## Description

Fixes https://github.com/vaadin/quarkus/issues/47

## Type of change

- [ ] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: articles/flow/integrations/quarkus.asciidoc [hunk 1]
```diff
@@ -18,9 +18,11 @@ It is a project template with the necessary configuration and dependencies inclu
 
 This starter is also available with Gradle configuration in the https://github.com/vaadin/base-starter-flow-quarkus/tree/gradle[Gradle] branch.
 
+[[quarkus.dependencies]]
 == Adding Dependencies
 
-To be able to run Quarkus, you need to have the `vaadin-quarkus` and `vaadin-jandex` Maven dependencies in the project, as well as configure the `quarkus-maven-plugin`.
+https://github.com/vaadin/base-starter-flow-quarkus/[Quarkus base starter] has all the needed dependencies to start developing your application.
```
Comment (caalador) [inline articles/flow/integrations/quarkus.asciidoc:24]: This line is extra and the `Adding Dependencies` should be renamed to `Manual Setup` or something as just before this "chapter" there is the line `It is a project template with the necessary configuration and dependencies included for starting to build your own application.` saying exactly this. It shouldn't need duplicating.

File: articles/flow/integrations/quarkus.asciidoc [hunk 2]
```diff
@@ -137,6 +130,68 @@ quarkus.index-dependency.<name>.artifact-id=
...
+                    </execution>
+                </executions>
+                <configuration>
+                    <productionMode>true</productionMode>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</profile>
+----
```
Comment (caalador) [inline articles/flow/integrations/quarkus.asciidoc:187]: This is the default production mode profile and should just link to https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode with the addition of if you didn't start from the starter.

**Review by caalador — CHANGES_REQUESTED** [review]

File: articles/flow/integrations/quarkus.asciidoc [hunk 3]
```diff
@@ -137,6 +130,68 @@ quarkus.index-dependency.<name>.artifact-id=
...
 ----
 
+The `<name>` string here is used to link the `group-id`, `artifact-id` and `classifier` entries in one logical block.
+It should be the same for those three entries and be any string literal.
+
+== Running the Application in Development Mode
+
+After making <<quarkus.dependencies>> steps, the Quarkus application can be started in development mode using the `quarkus:dev` goal in Maven.
+
+[source,bash]
```
Comment (jouni) [inline articles/flow/integrations/quarkus.asciidoc:140]: Let’s use "terminal" as the general term for command line snippets.

```suggestion
[source,terminal]
```

Comment (mshabarov) [reply articles/flow/integrations/quarkus.asciidoc]: done

File: articles/flow/integrations/quarkus.asciidoc (hunk 2 above)
Comment (mshabarov) [reply articles/flow/integrations/quarkus.asciidoc]: done

File: articles/flow/integrations/quarkus.asciidoc (hunk 1 above)
Comment (mshabarov) [reply articles/flow/integrations/quarkus.asciidoc]: done

File: articles/flow/integrations/quarkus.asciidoc [hunk 4]
```diff
@@ -146,47 +145,11 @@ The application is then available at http://localhost:8080/[+localhost:8080+] in
 
 == Running the Application in Production Mode
 
-If you want to run your application locally in production mode, add the following `production` profile to your project:
+Quarkus base starter already includes the needed maven configuration for running application in production mode.
```
Comment (jouni) [inline articles/flow/integrations/quarkus.asciidoc:148]:
```suggestion
Quarkus base starter already includes the needed Maven configuration for running the application in production mode.
```

File: articles/flow/integrations/quarkus.asciidoc [hunk 5]
```diff
@@ -137,6 +129,32 @@ quarkus.index-dependency.<name>.artifact-id=
...
+
+Quarkus base starter already includes the needed Maven configuration for running the application in production mode.
+If you use a project not based on the starter, it needs the configuration described in https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode[Enabling the Production Mode].
+
+Run the following commands to start the application:
+
+```
+mvn package -Pproduction
+java -jar target/quarkus-app/quarkus-run.jar
+```
```
Comment (caalador) [inline articles/flow/integrations/quarkus.asciidoc:156]: To keep the same format as used before, should we have this as
```suggestion
[source,terminal]
----
$ mvn package -Pproduction
$ java -jar target/quarkus-app/quarkus-run.jar
----
```

Comment (jouni) [reply articles/flow/integrations/quarkus.asciidoc:156]: I think we shouldn't use the `$` symbol in terminal commands. Doesn't that make it difficult to copy paste?

Comment (caalador) [reply articles/flow/integrations/quarkus.asciidoc:156]: That would be good too. I expect the whole documentation should be looked through to not have them even for single line commands.

Comment (jouni) [reply articles/flow/integrations/quarkus.asciidoc:156]: I’ll unify these in the latest branch.

Comment (jouni) [reply articles/flow/integrations/quarkus.asciidoc:156]: Updated everything else except this Quarkus article.

File: articles/flow/integrations/quarkus.asciidoc [hunk 6]
```diff
@@ -137,6 +129,33 @@ quarkus.index-dependency.<name>.artifact-id=
...
+----
+$ mvn package quarkus:dev
+----
+
+The application is then available at http://localhost:8080/[+localhost:8080+] in the browser.
+
+== Running the Application in Production Mode
+
+Quarkus base starter already includes the needed Maven configuration for running the application in production mode.
+If you use a project not based on the starter, it needs the configuration described in https://vaadin.com/docs/latest/flow/guide/production/#enabling-the-production-mode[Enabling the Production Mode].
```
Comment (jouni) [inline articles/flow/integrations/quarkus.asciidoc:149]:
```suggestion
If you use a project not based on the starter, it needs the configuration described in <<{articles}/flow/guide/production#enabling-the-production-mode, Enabling the Production Mode>>.
```

File: articles/flow/integrations/quarkus.asciidoc [hunk 7]
```diff
@@ -137,6 +129,33 @@ quarkus.index-dependency.<name>.artifact-id=
...
+The `<name>` string here is used to link the `group-id`, `artifact-id` and `classifier` entries in one logical block.
+It should be the same for those three entries and be any string literal.
+
+== Running the Application in Development Mode
+
+After making <<quarkus.setup>> steps, the Quarkus application can be started in development mode using the `quarkus:dev` goal in Maven.
+
+[source,terminal]
+----
+$ mvn package quarkus:dev
```
Comment (jouni) [inline articles/flow/integrations/quarkus.asciidoc:141]:
```suggestion
mvn package quarkus:dev
```

**Review by caalador — APPROVED** [review]

**Pull request merged**